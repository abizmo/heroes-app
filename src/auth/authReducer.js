import { LOGIN, LOGOUT } from "./authActionTypes"

const authReducer = (state = {}, { type, payload }) => {
  switch (type) {
    case LOGIN:
      return { ...payload, logged: true }

    case LOGOUT:
      return { logged: false }

    default:
      return state
  }
}

export default authReducer
