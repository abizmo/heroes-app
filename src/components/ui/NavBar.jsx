import React, { useContext } from 'react'
import { useHistory } from 'react-router'
import { Link, NavLink } from 'react-router-dom'
import { LOGOUT } from '../../auth/authActionTypes'
import AuthContext from '../../auth/AuthContext'

const NavBar = () => {
  const { user: { username }, dispatch } = useContext(AuthContext)
  const history = useHistory()

  const handleLogout = () => {
    dispatch({
      type: LOGOUT,
    })
    history.replace('/login')
  }

  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <Link
        className="navbar-brand"
        to="/"
      >
        Heroes
      </Link>

      <div className="navbar-collapse">
        <div className="navbar-nav">
          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/marvel"
          >
            Marvel
          </NavLink>

          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/dc"
          >
            DC
          </NavLink>

          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/search"
          >
            Search
          </NavLink>
        </div>
      </div>

      <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 justify-content-end">
        <ul className="navbar-nav ml-auto">
          <span className="nav-item nav-link text-info">{username}</span>
          <button
            className="nav-item nav-link btn btn-outline-secondary"
            onClick={handleLogout}
          >
            Logout
          </button>
        </ul>
      </div>
    </nav>
  )
}

export default NavBar