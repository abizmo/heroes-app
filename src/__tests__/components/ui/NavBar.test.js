import { mount } from "enzyme"
import { MemoryRouter, Router } from "react-router"
import { LOGOUT } from "../../../auth/authActionTypes"
import AuthContext from "../../../auth/AuthContext"
import NavBar from "../../../components/ui/NavBar"

describe('NavBar Tests', () => {
  const user = {
    username: 'goccita',
    logged: true,
  }
  const dispatch = jest.fn()
  const historyMock = {
    createHref: jest.fn(),
    replace: jest.fn(),
    location: {
    },
    listen: jest.fn(),
  }

  afterEach(() => {
    jest.clearAllMocks()
  })

  const wrapper = mount(
    <AuthContext.Provider value={{ user, dispatch }}>
      <MemoryRouter>
        <Router history={historyMock}>
          <NavBar />
        </Router>
      </MemoryRouter>
    </AuthContext.Provider>
  )

  test('should render and match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.text-info').text()).toBe(user.username)
  })
  
  test('should logout', () => {
    wrapper.find('button').simulate('click')

    expect(dispatch).toHaveBeenCalledWith({type: LOGOUT})
    expect(historyMock.replace).toHaveBeenCalledWith('/login')
  })
})
