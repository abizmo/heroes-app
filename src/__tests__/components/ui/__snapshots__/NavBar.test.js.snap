// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Tests should render and match snapshot 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Heroes
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2 justify-content-end"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        goccita
      </span>
      <button
        className="nav-item nav-link btn btn-outline-secondary"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
